lib/compress.js

```diff
@@ -4655,4 +4655,36 @@ merge(Compressor.prototype, {
     OPT(AST_PrefixedTemplateString, function(self, compressor){
         return self;
     });
+
+    OPT(AST_ObjectKeyVal, function(self, compressor){
+        // p:function(){} ---> p(){}
+        // p:function*(){} ---> *p(){}
+        // p:async function(){} ---> async p(){}
+        // p:()=>{} ---> p(){}
```

Comment:
Clearly this PR ought to avoid arrow functions altogether if they contain `this`.